components/brave_wallet_ui/components/desktop/popup-modals/confirm-password-modal/confirm-password-modal.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// utils
+import { getLocale } from '../../../../../common/locale'
+
+// hooks
+import { usePasswordAttempts } from '../../../../common/hooks/use-password-attempts'
+
+// components
+import { PopupModal } from '../..'
+import { NavButton } from '../../../extension'
+import { PasswordInput } from '../../../shared'
+
+// style
+import {
+  Row,
+  VerticalSpace
+} from '../../../shared/style'
+import { Title } from '../style'
+import {
+  modalWidth,
+  StyledWrapper
+} from './confirm-password-modal.style'
+
+interface Props {
+  title: string
+  onClose: () => void
+  onSuccess: () => void
+}
+
+export const ConfirmPasswordModal = ({
+  onClose,
+  onSuccess,
+  title
+}: Props) => {
+  // state
+  const [password, setPassword] = React.useState<string>('')
+  const [isCorrectPassword, setIsCorrectPassword] = React.useState<boolean>(true)
+
+  // custom hooks
+  const { attemptPasswordEntry } = usePasswordAttempts({ maxAttempts: 3 })
+
+  // methods
+  const onSubmit = React.useCallback(async () => {
```

Comment:
is there any way to add tests for this? ex: to make sure it doesn't work unless the password is correct, that too many attempts fail, and that it fails if attemptPasswordEntry throw an error.

Reply:
Added tests here: [83b1616](https://github.com/brave/brave-core/pull/14669/commits/83b1616505cb8683e212f31bd97fa774e31c4fe1)